src/FormBuilder/resources/js/form-builder/src/common/stripTags.ts

```diff
@@ -0,0 +1,5 @@
+export default function stripTags(text: string): string {
```

Comment:
This is cool approach to accomplishing the removal of tags,  is it possible to avoid creating dom elements? I tried a couple tags with a regex pattern `` `input.replace(/<[^>]*>/g, '');``` with some success but wanted to know your thoughts about this?

Reply:
```js

Reply:
TL;DR Try this:

```js
function stripHtml(dirtyString) {
  const doc = new DOMParser().parseFromString(dirtyString, 'text/html');
  return doc.body.textContent || '';
}
```

--

Chris Coyier has a snippet on CSS-Tricks for stripping tags using regex.

https://css-tricks.com/snippets/javascript/strip-html-tags-in-javascript/

That said, the comments suggest that the regex has issues with special characters (I have not tested this).

It is also noted that using the DOM to render the HTML and then pull `textContent` is open to an XSS attack, since the script tags are executed.

Alternatively, it is suggested that using `DOMParser()` will not execute the script tags, which is safer.

```js
function stripHtml(dirtyString) {
  const doc = new DOMParser().parseFromString(dirtyString, 'text/html');
  return doc.body.textContent || '';
}

// tripHtml("<img onerror="alert(\"could run arbitrary JS here\")"" src="bogus">"); 
```

Reply:
It does look good. I have tested and implement now.